File: Day_6/Learn-and-Test-Android-App-master/app/src/main/java/com/example/projectjava/MainActivity.java
package com.example.projectjava;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {


    FirebaseAuth mAuth;
    public CardView c1, c2, c3, c4;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        c1 = (CardView) findViewById(R.id.c1tutorial);
        c2 = (CardView) findViewById(R.id.c2interview);
        c3 = (CardView) findViewById(R.id.c3quiz);
        c4 = (CardView) findViewById(R.id.c3video);

        c1.setOnClickListener(this);
        c2.setOnClickListener(this);
        c3.setOnClickListener(this);
        c4.setOnClickListener(this);


        mAuth = FirebaseAuth.getInstance();



    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu1) {
        getMenuInflater().inflate(R.menu.menu,menu1);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        Intent i;
        switch (item.getItemId()){
            case R.id.profile:
                i = new Intent(this, profile.class);
                startActivity(i);
                break;


            case R.id.aboutus:
                i = new Intent(this, aboutus.class);
                startActivity(i);
                break;


            case R.id.logout1:
                mAuth.signOut();
                startActivity(new Intent(MainActivity.this, LoginActivity.class));
                break;

        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onStart() {
        super.onStart();
        FirebaseUser user = mAuth.getCurrentUser();
        if (user == null) {
            startActivity(new Intent(MainActivity.this, LoginActivity.class));
        }
    }

    @Override
    public void onClick(View view) {
        Intent i;
        switch (view.getId()) {
            case R.id.c1tutorial:
                i = new Intent(this, javatutorial.class);
                startActivity(i);
                break;

            case R.id.c2interview:
                i = new Intent(this, javainterview.class);
                startActivity(i);
                break;
            case R.id.c3quiz:
                i = new Intent(this, javaquiz.class);
                startActivity(i);
                break;
            case R.id.c3video:
                i = new Intent(this, javavideos.class);
                startActivity(i);
                break;

        }
    }
}

File: Day_5/app/src/main/java/com/wira_sukma_saputra/botanikacare/MainActivity.kt
package com.wira_sukma_saputra.botanikacare

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.Firebase
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.auth
import com.wira_sukma_saputra.botanikacare.about.AboutActivity
import com.wira_sukma_saputra.botanikacare.databinding.ActivityMainBinding
import com.wira_sukma_saputra.botanikacare.splashscreen.GetStartedActivity


class MainActivity : AppCompatActivity() {
    private var binding: ActivityMainBinding? = null
    private lateinit var auth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding?.root)
        auth = Firebase.auth

        binding?.SignOut?.setOnClickListener{
            if (auth.currentUser!=null)
            {
                auth.signOut()
                startActivity(Intent(this, GetStartedActivity::class.java))
                finish()
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_item, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.menu_about -> {
                startActivity(Intent(this, AboutActivity::class.java))
                return true
            }
            R.id.action_list -> {
                startActivity(Intent(this, MapsActivity::class.java))
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }
}